helix-core/src/main/java/org/apache/helix/model/CustomizedView.java

```diff
@@ -0,0 +1,122 @@
+package org.apache.helix.model;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
+
+import org.apache.helix.HelixProperty;
+import org.apache.helix.ZNRecord;
+
+/**
+ * Customized view is an aggregation (across all instances)
+ * of customized states for the resource
+ */
+public class CustomizedView extends HelixProperty {
+
+  /**
+   * Properties that are persisted and are queryable for a customized view
+   */
+  public enum CustomizedViewProperty {
+    STATE_MODEL_DEF_REF,
```

Comment:
I think we discussed that we don't need a state model?
And for replicas, not sure how our customer will be using it. Could you please comment?

Reply:
You are right. We do not need these two fields for now. I removed them.